From 63d3c2f252ec8a292f4282a54ec035e45c20e730 Mon Sep 17 00:00:00 2001
From: Steven Weaver <sweaver@temple.edu>
Date: Sun, 29 Oct 2023 11:38:47 -0600
Subject: [PATCH] remove darwin check

This patch removes the platform check for Darwin (macOS) systems in `setup.py` during the build extension process.

Rationale:
- Pysam's setup process includes specific checks and configurations for macOS that are not applicable when cross-compiling for WebAssembly using Emscripten.
- The check for `sys.platform == 'darwin'` is replaced with `None` to bypass macOS-specific configurations, as these are irrelevant and potentially problematic in a cross-compilation context.
- This ensures that the build process does not attempt to apply macOS-specific build settings that are incompatible with the target WebAssembly environment.

This modification is necessary to avoid build errors and misconfigurations when compiling Pysam for an Emscripten environment rather than a native macOS environment.

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 39da16d..9f3714f 100644
--- a/setup.py
+++ b/setup.py
@@ -275,7 +275,7 @@ class cy_build_ext(build_ext):
         if not self.inplace:
             ext.library_dirs.append(os.path.join(self.build_lib, "pysam"))
 
-        if sys.platform == 'darwin':
+        if None:
             # The idea is to give shared libraries an install name of the form
             # `@rpath/<library-name.so>`, and to set the rpath equal to
             # @loader_path. This will allow Python packages to find the library
-- 
2.39.3 (Apple Git-145)

